### DISCUSSION TITLE
Statically link image magic programmes on Linux

### DISCUSSION THREAD

**Discussion by WorikWork** [ImageMagick/ImageMagick#7347] [Help]
### ImageMagick version

8:6.9.11.60+dfsg-1.6+deb12u1

### Operating system, version and so on

Linux Debian 11

### Description

Friends

I am building a little application in C to scale images.  Very simple and the the Image Magick libraries have been hugely helpful.

I need to compile a static version of my programme so I can avoid installing extra libraries on my server for this one application

I am compiling it at the moment: `gcc scale_jpeg.c -o scale_jpeg -I/usr/include/x86_64-linux-gnu/ImageMagick-6 -I/usr/include/ImageMagick-6 -fopenmp -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -lMagickWand-6.Q16 -lMagickCore-6.Q16`

This produces a dynamically linked executable which is very good

My gcc foo is low, so I am unsure how to proceed

Naively I tried adding `-static` and `-static-libgcc` but clearly that is not enough.

I have looked through the forums here and I only see people wanting to build Image Magic itself as static.  If my question has been asked and answered I am sorry, a link would be very helpful.

**Comment by urban-warrior** [comment]
You can build a static version of ImageMagick with:

```
$ ./configure --disable-shared
$ make
$ make install
```

Now build your program as static and link to the static version of ImageMagick.

**Comment by WorikWork** [comment]
> $ ./configure --disable-shared
> $ make
> $ make install

To be perfectly clear:  To make static executables utilising Image Magic I need to build from source?

I am on Debian 11, utilising Image Magic from the repositories.  

```bash
$ aptitude search imagemag|grep '^i '
i A imagemagick - image manipulation programs -- binaries
i A imagemagick-6-common - image manipulation programs -- infrastructure
i A imagemagick-6.q16 - image manipulation programs -- quantum depth Q16
```

There is no way I can  build static executables utilising those packages?
